utxo/src/cache.rs

```diff
@@ -145,6 +153,73 @@ impl<'a> UtxosCache<'a> {
         tx_undo.map(TxUndo::new)
     }
 
+    // Marks outputs of a transaction as spent and inputs as unspent
+    pub fn unspend_utxos_from_tx(
+        &mut self,
+        tx: &Transaction,
+        tx_undo: &TxUndo,
+    ) -> Result<(), Error> {
+        for (i, _) in tx.outputs().iter().enumerate() {
+            let tx_outpoint = OutPoint::new(OutPointSourceId::from(tx.get_id()), i as u32);
+            self.spend_utxo(&tx_outpoint)?;
+        }
+
+        for (tx_in, utxo) in tx.inputs().iter().zip(tx_undo.inner().iter()) {
+            self.add_utxo(tx_in.outpoint(), utxo.clone(), false)?;
+        }
+        Ok(())
+    }
+
+    /// Marks the inputs of a transactable block reward as 'spent', adds outputs to the utxo set.
+    /// If BlockRewardTransactable has no inputs then just adds outputs to utxo set.
+    /// Returns a BlockRewardUndo if function is a success or an error if the input cannot be spent.
+    pub fn spend_utxos_from_block_transactable(
```

Comment:
I think the right terminology here is "connect_block_transactable". Spending is the act of marking an input as spent and/or removing it from the utxo set. Here you're doing both, spending inputs and adding outputs, we call that connecting.